src/index.js

```diff
@@ -0,0 +1,34 @@
+const { formatDate, isDateInList, bankHolidayList, validCountry } = require('./utils');
+const request = require('superagent');
+
+const isDateABankHoliday = async function(date, countries) {
+
+    if (!date && !countries) {
+        throw new Error('No arguments passed');
+    }
+
+    if (!countries) {
+        throw new Error('Countries not passed');
+    }
+
+    if (!Array.isArray(countries)) {
+        throw new Error('Countries must be an array');
+    }
+
+    if (!validCountry(countries)) {
+        throw new Error('Country not on the list');
+    }
+
+    date = formatDate(date);
+
+    try {
+        const bankHolidays = await request.get('https://www.gov.uk/bank-holidays.json');
```

Comment:
Going forward we should think about cacheing this call, maybe for 24hrs.

Reply:
Something like this? https://www.npmjs.com/package/memory-cache

Reply:
This appears to be more popular: https://www.npmjs.com/package/node-cache